cranelift/codegen/src/machinst/mod.rs

```diff
@@ -269,6 +272,48 @@ pub trait MachInstLabelUse: Clone + Copy + Debug + Eq {
     fn from_reloc(reloc: Reloc, addend: Addend) -> Option<Self>;
 }
 
+/// Classification of call instruction types for granular analysis.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum CallType {
```

Comment:
Can we make these enums `pub(crate)`? There's no reason to export them to the world (we don't support external backends).

Reply:
I agree for `FunctionType` / `FunctionCalls` enum.

But for `CallType`, since I added to the `pub trait MachInst` a `call_type()` method returning `CallType` I need to either keep it public or add `#[allow(private_interface)]` to the method